ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -233,4 +233,22 @@ message DisclosedContract {
   // The contract metadata from the create event.
   // Required
   ContractMetadata metadata = 4;
+
+  // The signatories for this contract as specified by the template.
+  // Required
+  repeated string signatories = 6;
+
+  // The observers for this contract as specified explicitly by the template or implicitly as choice controllers.
+  // This field never contains parties that are signatories.
+  // Optional
+  repeated string observers = 7;
+
+  // The key of the created contract.
+  // This will be set if and only if ``create_arguments`` is set and ``template_id`` defines a contract key.
+  // Optional
+  Value contract_key = 8;
+
+  // Maintainers of the contract key, if defined.
+  // Optional
+  repeated string key_maintainers = 9;
```

Comment:
@tudor-da : I'm probably missing context, but why do we need all of these additional fields? This means that every third-party store (think PQS driven by scribe, or the [persistent stores used by CN apps](https://docs.google.com/document/d/1kF3jkPm_seKnL-tnkMfIE_hjIVIWy4lPVHgQk-c_pLs/edit#heading=h.j1o9vy5fqmrz)) will have to store and retrieve these fields as well, even though they can be computed by the Ledger API server.

It is currently only an intuition of mine that we want to minimize the replication of computed fields, but I'm wary of this direction. @cocreature wdyt?